### DISCUSSION TITLE
How do I change the device name on the network?

### DISCUSSION THREAD

**Discussion by jingqunhu** [nanoframework/Home#1531] [Q&A]
I see the list of devices on the network through the router, and our nano displays the following: nanodevice_******, is there any way to change the device name?

**Comment by alberk8** [comment]
@jingqunhu,  It is in the native code and you can try to compile your own firmware from [source](https://github.com/nanoframework/nf-interpreter/blob/6cd9a722da6dad2062d8e07e993fdcb868cd8180/targets/ESP32/_common/targetHAL_Network.cpp#L28).

**Reply by bsod79** [reply]
@jingqunhu It depends on your need. If you need a smart way to reach it through browser you can you MulticastDNS to assigna  name to your device.
This is my sample code:


        
public class mDNSManager
{
        MulticastDnsService multicastDNSService = null;
        public void Init()
        {
            multicastDNSService = new MulticastDnsService();
            multicastDNSService.MessageReceived += MulticastDNSService_MessageReceived;
        }

        private void MulticastDNSService_MessageReceived(object sender, MessageReceivedEventArgs e)
        {
            var q = e.Message.GetQuestions();

            Debug.WriteLine($"Multicast DNS message received: Total question: {e.Message.GetQuestions().Length.ToString()}");
            foreach (var question in q)
            {
                Debug.WriteLine($"Question: {question.Domain}, Type: {question.QueryType}, Class: {question.QueryClass}");

                if (question.QueryType == Iot.Device.MulticastDns.Enum.DnsResourceType.A)
                {
                    if (question.Domain == "<custom name 1>.local" || 
                        question.Domain == "<custom name 1>.local" || 
                        question.Domain == "nanoff.local" )
                    {
                        Debug.WriteLine($"Generating response for {question.Domain}");

                        // Get the IP address of the device
                        var ipAddress = NetInfo.CurrentIPAddress;
                        
                        if (ipAddress != null)
                        {
                            e.Response = new Iot.Device.MulticastDns.Entities.Response();
                            var res = new Iot.Device.MulticastDns.Entities.ARecord(question.Domain, ipAddress, 120);
                            
                            e.Response.AddAnswer(res);                                                        

                            Debug.WriteLine($"Response sent for {question.Domain} with IP {ipAddress.ToString()}");
                        }
                        else
                        {
                            Debug.WriteLine("No IP address found for this device.");
                        }
                    }
                }
            }
        }

        

        public void Start()
        {
            Debug.WriteLine("Starting mDNS service...");
            multicastDNSService.Start();
            Debug.WriteLine("mDNS service started.");
        }
        public void Stop()
        {
            Debug.WriteLine("Stopping mDNS service...");
            multicastDNSService.Stop();
            Debug.WriteLine("mDNS service stopped.");
        }

        ~mDNSManager()
        {
            Stop();
            Debug.WriteLine("mDNSManager finalized and stopped.");
        }
}

Hope this helps

**Comment by josesimoes** [comment]
If this is with Wi-FI, this has just been added: nanoframework/System.Device.Wifi#322
